crypto/defaults.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 1995-2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include "internal/thread_once.h"
+#include "internal/cryptlib.h"
+#include "internal/e_os.h"
+
+#if defined(_WIN32)
+
+/**
+ * @brief The directory where OpenSSL is installed.
+ */
+static char openssldir[MAX_PATH];
```

Comment:
It's `static`, so it should be initialized at build time to zero (according to C11)

Reply:
Yes, this did occur to me - but it caused me pause for thought to remind myself what the C spec said about such things (I *think* this applies to C90 too, but I'm not sure). But it might be nicer to make it explicit anyway if we're going to rely on it.

Reply:
I would say this is a well known C rule and we depend on it in other cases as well. AFAIK it is there all the way down to K&R C specs. However adding explicit initializer wouldn't hurt.